src/GafferSceneUI/ContextAlgo.cpp

```diff
@@ -100,42 +102,45 @@ namespace ContextAlgo
 
 void setExpandedPaths( Context *context, const IECore::PathMatcher &paths )
 {
-	context->set( g_expandedPathsName, new IECore::PathMatcherData( paths ) );
+	auto visibleSet = context->get<VisibleSet>( g_visibleSetName, VisibleSet() );
+	visibleSet.expansions = paths;
+	context->set( g_visibleSetName, new VisibleSetData( visibleSet ) );
```

Comment:
I'm pretty sure that `new VisibleSetData` is leaked here. `Context::set()` takes a raw `Data *` because it doesn't want to take ownership of the original argument (it makes a copy instead), so nobody ever takes ownership and we have a leak. I think we can do `context->set( g_visibleSetName, visibleSet )` instead, and then only one VisibleSetData is allocated (made internally and owned by the context). Better still, we could use `getVisibleSet()` and `setVisibleSet()` here, so the actual setting logic remains encapsulated in those...

Reply:
Ah, nice catch! I've adjusted that in d4e9c57 and tried to use `setVisibleSet()` and `getVisibleSet()` wherever possible.